src/templates/cmp.js

```diff
@@ -213,4 +223,56 @@
   } else {
     loadTcfapi(3);
   }
+
+  // send device ids
+  function sendDeviceId(consent, retriesLeft) {
+    if (retriesLeft < 0) {
+      return;
+    }
+
+    if (!__hbb_tracking_tgt || !__hbb_tracking_tgt.getDID) {
+      setTimeout(function () {
+        sendDeviceId(consent, retriesLeft - 1);
+      }, 100);
+      return;
+    }
+
+    __hbb_tracking_tgt.getDID(function (deviceId) {
+      var image = document.createElement('img');
+      image.src =
+        '<%-TRACKING_PROTOCOL%>://' +
+        (consent ? '<%-TRACKING_HOST_CONSENT%>' : '<%-TRACKING_HOST_NO_CONSENT%>') +
+        '/<%-TRACKING_VERSION%>/consent-status/' +
+        deviceId +
+        '/' +
+        Date.now() +
+        '/consent.gif?consent=' +
+        consent;
+
+      image.onload = function () {
+        window.__tcfapi(
+          'log',
+          2,
+          undefined,
+          '{"event":"onConsent","success":true,"parameters":{"consent":' + consent + '}}',
+        );
+      };
+      image.onerror = function () {
+        window.__tcfapi(
+          'log',
+          2,
+          undefined,
+          '{"event":"onConsent","success":false,"parameters":{"consent":' + consent + '}}',
+        );
+      };
+    });
+  }
+
+  window.__tcfapi('onConsent', 2, function (consent) {
```

Comment:
Wouldn't it be sufficient to subscribe on the log API to get triggered?

Reply:
No, reason for this is that the log api can currently only handle one callback. meaning if you set a callback, any previous callback wouldn't work anymore. Therefore I added a new method specifically for this purpose.